core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +109,38 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory  in the scm
+     * @param pluginName the name of the plugin
+     * @param scm        the valid scm link
+     * @return folderPath if it valid or return the scm itself
+     */
+    private String searchPomFiles(Path directory, String pluginName, String scm) {
+        try (Stream<Path> paths = Files.find(directory, 2, (path, $) ->
+            path.getFileName().toString().equals("pom.xml")
+        )) {
```

Comment:
just for readability

```suggestion
        try (Stream<Path> paths = Files.find(directory, 2, (path, $) -> path.getFileName().toString().equals("pom.xml"))) {
```